microprofile/WFLY_13129_Microprofile_CLI_Script_Example.adoc

```diff
@@ -0,0 +1,95 @@
+= Example CLI script to evolve standalone configurations with microprofile
+:author:            JF Denise
+:email:             jdenise@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+We are proposing here to define a new script in docs/examples/enable-microprofile.cli
+The script is an “example” script to be applied on default standalone configurations. 
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFLY-13129
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* TBD
+
+=== Testing By
+
+[ ] Engineering
+
+[ ] QE
+
+=== Affected Projects or Components
+
+* WildFly
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+* The script is packaged in the galleon-pack maven module. It is a default package installed when WildFly is installed.
+* Installation path is _<JBOSS_HOME>/docs/examples/enable-microprofile.cli_.
+* To launch the script: _bin/jboss-cli.sh --file=docs/examples/enable-microprofile.cli_
+* By default it applies to standalone.xml, use _-Dconfig=<standalone-full.xml|standalone-ha.xml|standalone-full-ha.xml>_
+to apply to other standalone configurations.
+* The script informs the user with high level information on the modifications it applies to the configuration it is updating.
+* The script is expected to adapt to the differences that exist between various standalone configurations.
+* The script detects that a microprofile subsystem is already present in the config and skips it. 
+* The script secures the configuration using elytron security domain ApplicationDomain, http-authentication-factory and sasl-authentication-factory for items present in default configurations:
+** core-service/management
+*** access/identity security domain
+*** managementItf/http-upgrade sasl factory
+*** managementItf/ http auth factory
+** remoting/http-connector sasl factory
+** undertow 
+*** other application-domain
+*** http-invoker UNTOUCHED, references legacy security realm, no http-auth factory to set.
```

Comment:
If the only thing missing is the http-auth-factory we can add that within a script.

Reply:
@darranl , the question I have is what should be the content of this http-auth-factory? Should we add back the http-auth-factory that was existing at some point?

Reply:
Yes I think we should.

A bit off topic for this feature but I think maybe we should also revisit at some point the resources that reference a factory and see if we can do direct security domain references maybe with some additional attributes to select the available authentication mechanisms.

Reply:
@darranl , the factory that we could add and use for http-invoker:
```
<http-authentication-factory name="application-http-authentication" http-server-mechanism-factory="global" security-domain="ApplicationDomain">
  <mechanism-configuration>
    <mechanism mechanism-name="BASIC">
      <mechanism-realm realm-name="Application Realm"/>
    </mechanism>
    <mechanism mechanism-name="FORM"/>
 </mechanism-configuration>
</http-authentication-factory>
```

Reply:
As this is now only being used for the invoker I would say drop FORM form the list of mechanisms so it is BASIC only

Reply:
@darranl , I will mark this possible evolution as "Future work", it seems to go beyond the scope of this example script that is not expected to optimize fully the dependency on Realms.